### PR TITLE
Notification trigger and template creation.

### PR DESCRIPTION
**Pull request by kcherukupalli**
## What’s changing?

 Create notification trigger logic and create Match creation notification template(type 1)

## Why?

@nac-1099
@nac-1115

## This PR has:

- [ x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

File: notifications/src/Piipan.Notification.Common/Models/DispositionModel.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+﻿namespace Piipan.Notification.Common.Models
+{
+    public class DispositionModel
+    {
+        //TODO for nac-1115 Type 2 Notifications
```
Comment (bgfoshay) [inline notifications/src/Piipan.Notification.Common/Models/DispositionModel.cs:5]: Nitpick: can this comment be removed?

File: notifications/src/Piipan.Notifications.Core/Builders/NotificationRecordBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        /// Get the built record and reset internal record reference.
+        /// </summary>
+        /// <returns>Current NotificationRecord instance</returns>
+        public NotificationRecord GetRecord()
+        {
+            NotificationRecord record = this._record;
+
+            this.Reset();
+
+            return record;
```
Comment (bgfoshay) [inline notifications/src/Piipan.Notifications.Core/Builders/NotificationRecordBuilder.cs:27]: Question: Does this return the built record or a new NotificationRecord each time? I may be reading incorrectly, but the assignment above looks like a reference assignment and if you reset it, won't it end up modifying the underlying object?

File: notifications/src/Piipan.Notifications.Core/Extensions/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using Piipan.Notification.Common;
+using Piipan.Notifications.Core.Builders;
+
+namespace Piipan.Notification.Core.Extensions
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static void RegisterNotificationServices(this IServiceCollection serviceCollection)
+        {
+            //serviceCollection.TryAddTransient<TokenCredential, ManagedIdentityCredential>();
```
Comment (bgfoshay) [inline notifications/src/Piipan.Notifications.Core/Extensions/ServiceCollectionExtensions.cs:13]: Nitpick: Should these comments be removed?

File: notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs [hunk 4]
```diff
@@ -1,80 +1,57 @@
...
-        public NotificationService(INotificationPublish notificationPublish, ILogger<NotificationService> logger)
+        public NotificationService(INotificationPublish notificationPublish, IViewRenderService viewRenderService, ILogger<NotificationService> logger)
         {
             _notificationPublish = notificationPublish;
+            _viewRenderService = viewRenderService;
             _logger = logger;
         }
 
-        public async Task<bool> PublishMessageFromTemplate(EmailTemplateInput emailTemplateInput)
+        public async Task<bool> PublishNotificationOnMatchCreation(NotificationRecord notificationRecord)
```
Comment (bgfoshay) [inline notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs:22]: Thought: Could we consider consolidating this to publishing a single email? Then update MatchEventService to provide the cshtml file names? e.g. for IS, provide "MatchEmailIS.cshtml" & "MatchEmailIS_Sub.cshtml"

Comment (bgfoshay) [reply notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs:22]: So there would be two calls to PublishNotificationOnMatchCreation in MatchEventService

Comment (kcherukupalli) [reply notifications/src/Piipan.Notifications.Core/Services/NotificationService.cs:22]: Refactored to only one call for PublishNotificationOnMatchCreation, which includes Emailto for both Matching and Initiating state.

File: match/tests/Piipan.Match.Func.ResolutionApi.Tests/AddEventApiTests.cs [hunk 5]
```diff
@@ -743,5 +842,181 @@ public async void AddEvent_CheckIfFinalDispositionDateError()
             ), Times.Never());
             Assert.Equal(400, response.StatusCode);
         }
+        [Fact]
+        public async void ValidateForNotificationServiceTest()
```
[the remark this answers is not in the document]
Comment (kcherukupalli) [reply match/tests/Piipan.Match.Func.ResolutionApi.Tests/AddEventApiTests.cs:846]: Tried to have the entire business logic for type 2 notification trigger in one test case.  Let me know if you still want this to be split in multiple test cases.

File: notifications/src/Piipan.Notifications.Core/Builders/NotificationRecordBuilder.cs (hunk 2 above)
Comment (kcherukupalli) [reply notifications/src/Piipan.Notifications.Core/Builders/NotificationRecordBuilder.cs:27]: it returns the built record,  we return the local variable and reset the this._record.

File: notifications/src/Piipan.Notifications.Core/Builders/NotificationRecordBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,92 @@
+﻿using Piipan.Notification.Common.Models;
+using Piipan.Notifications.Models;
+
+namespace Piipan.Notifications.Core.Builders
+{
+    public class NotificationRecordBuilder : INotificationRecordBuilder
```
Comment (bgfoshay) [inline notifications/src/Piipan.Notifications.Core/Builders/NotificationRecordBuilder.cs:6]: Question: Should there be a set of unit tests for this class? I didn't see NotificationRecordBuilderTests (or anything similarly named)?

**Review by bgfoshay — APPROVED** [review]
👍

**Pull request merged**